### ISSUE TITLE
Support remote kernel on ipv6 machine

### ISSUE THREAD

**Issue by wusanye** [jupyter/enterprise_gateway#929]
## Description
Hi, 

Does EG have a plan to support remote kernel on ipv6 machine?
I think the only thing to care about is zmq ipv6 option in KernelManager and IPKernelApp, like:
```
context.setsockopt(zmq.IPV6, 1)
```

I have checked the code of ipykernel and jupyter_client, found that they don't enable ipv6.

I know that this stuff is part of work of ipykernel and jupyter_client, but they don't have ipv6 problem due to local kernel. So I new a issue here.

**Comment by kevin-bates** [comment]
Yes, that would be the plan and have kept issue #428 open for that reason.  The response address used between EG and the remote kernels would also need to be made ipv6 aware - so each launcher would require updates as well.

This contribution would likely need to come from the community as EG's current maintainers don't have the bandwidth for this.  Is this something you'd like to help contribute?

**Comment by suryag10** [comment]
hi, 
we had done a POC and were successful in the same. Will publish the patch in sometime.

**Comment by kevin-bates** [comment]
Hi @suryag10, 

> we had done a POC and were successful in the same. Will publish the patch in sometime.

Are you able to share/contribute your findings?
